lib/contentprovider/xbmcprovider.py

```diff
@@ -43,6 +43,10 @@ def __init__(self,provider,settings,addon):
             provider.lang = None
             pass
         self.settings = settings
+        # lang setting is optional for plugins
+        if not 'lang' in self.settings.keys():
```

Comment:
Keby si pouzil `'lang' in self.settings`, tak ti ten check zbehne v O(log n). Ked ale robis `'lang' in self.settings.keys()`, tak ti to zbehne v O(n). Tj., nechces pouzivat `keys()` na checkovanie existencie kluca v dictionary.